### PR TITLE
added emergency space view

### PR DESCRIPTION
**Pull request by alexey-petrov**
Closes https://github.com/sparkletown/internal-sparkle-issues/issues/1017

### CODE REVIEW

File: src/pages/EmergencyViewPage/EmergencyViewPage.tsx [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+  { id: 0, title: "Spaces" },
+  { id: 1, title: `What's on` },
+];
+
+export const EmergencyViewPage: React.FC = () => {
+  const [selectedTab, updateTab] = useState(0);
+  const venueId = useVenueId() || "";
+  useConnectCurrentVenue();
+
+  const venue = useSelector(currentVenueSelectorData);
```
Comment (dmitrymarchuk) [inline src/pages/EmergencyViewPage/EmergencyViewPage.tsx:52]: won't be needed if `useConnectCurrentVenueNg` is used

File: src/pages/EmergencyViewPage/EmergencyViewPage.tsx [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+  }, [hostname, redirectUrl]);
+
+  dayjs.extend(advancedFormat);
+
+  useEffect(() => {
+    if (!venue) return;
+
+    // @debt replace this with useCss?
+    updateTheme(venue);
+  }, [venue]);
```
Comment (dmitrymarchuk) [inline src/pages/EmergencyViewPage/EmergencyViewPage.tsx:86]: This is repeated in a codebase. It needs to be extracted into a separate hook everywhere and not duplicated once more here. And maybe replaced with `useCss`:)

File: src/pages/EmergencyViewPage/EmergencyViewPage.tsx [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+  const redirectUrl = venue?.config?.redirectUrl ?? "";
+  const { hostname } = window.location;
+
+  useEffect(() => {
+    if (redirectUrl && redirectUrl !== hostname) {
+      window.location.hostname = redirectUrl;
+    }
+  }, [hostname, redirectUrl]);
+
+  dayjs.extend(advancedFormat);
```
Comment (dmitrymarchuk) [inline src/pages/EmergencyViewPage/EmergencyViewPage.tsx:79]: This does not need to be called inside a component body. Just needs to be called once at a top-level scope.

File: src/pages/EmergencyViewPage/EmergencyViewPage.tsx [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+              title={`Events on ${formatDateRelativeToNow(day)}`}
+              isShowBookmark={false}
+              isShowImage={false}
+              isShowDayDate={false}
+            />
+          </div>
+        );
+      })
+      .filter((day) => !!day);
+  }, [dayDifference, liveAndFutureEvents, firstScheduleDate]);
```
Comment (dmitrymarchuk) [inline src/pages/EmergencyViewPage/EmergencyViewPage.tsx:124]: Can be extracted into a separate utility component (including wrapping `<ul>` tag) for the `EmergencyViewPage`

File: src/pages/EmergencyViewPage/EmergencyViewPage.tsx [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+        const dailyEvents = liveAndFutureEvents.filter(
+          isEventWithinDateAndNotFinished(day)
+        );
+
+        const eventsFilledWithPriority = dailyEvents
+          .map((event) => ({
+            ...event,
+            orderPriority: event.orderPriority ?? 0,
+          }))
+          .sort(eventTimeAndOrderComparator);
```
Comment (dmitrymarchuk) [inline src/pages/EmergencyViewPage/EmergencyViewPage.tsx:102]: This is a duplicate of `NavBarSchedule.tsx` on line 144.

File: src/pages/EmergencyViewPage/EmergencyViewPage.tsx [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+      <div className="container venue-entrance-experience-container">
+        <div
+          className="header"
+          style={{
+            background: `linear-gradient(
+            0deg,
+            rgba(0, 0, 0, 0.8) 2%,
+            rgba(0, 0, 0, 0) 98%
+          ), url(${
+            venue.config?.landingPageConfig.bannerImageUrl ??
```
Comment (dmitrymarchuk) [inline src/pages/EmergencyViewPage/EmergencyViewPage.tsx:149]: can we use classes here?

File: src/pages/EmergencyViewPage/EmergencyViewPage.tsx [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+              />
+            </div>
+
+            <div className="title">{venue.name}</div>
+
+            <div className="subtitle">
+              {venue.config?.landingPageConfig.subtitle}
+            </div>
+          </div>
+        </div>
```
Comment (dmitrymarchuk) [inline src/pages/EmergencyViewPage/EmergencyViewPage.tsx:170]: This is a duplicate of the code in `VenueLandingPage.tsx` starting on lines 93.

File: src/pages/EmergencyViewPage/EmergencyViewPage.tsx [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+                  key={`venue-presentation-paragraph-${index}`}
+                  className="presentation-paragraph"
+                >
+                  {paragraph}
+                </p>
+              )
+            )}
+          </div>
+        </div>
+      </div>
```
Comment (dmitrymarchuk) [inline src/pages/EmergencyViewPage/EmergencyViewPage.tsx:234]: This is also duplicated from `VenueLandingPage.tsx`

File: src/pages/EmergencyViewPage/EmergencyViewPage.tsx [hunk 9]
```diff
@@ -0,0 +1,281 @@
...
+              className={`EmergencyView_tabs_item${
+                tab.id === selectedTab ? "--active" : ""
+              }`}
+              key={tab.id}
+              onClick={() => updateTab(tab.id)}
+            >
+              {tab.title}
+            </div>
+          ))}
+        </div>
```
Comment (dmitrymarchuk) [inline src/pages/EmergencyViewPage/EmergencyViewPage.tsx:248]: Can be extracted into a separate component.

File: src/pages/EmergencyViewPage/EmergencyViewPage.tsx [hunk 10]
```diff
@@ -0,0 +1,281 @@
...
+            )}
+          </div>
+        </div>
+      </div>
+      <div className="EmergencyView">
+        <div className="EmergencyView_tabs">
+          {tabs.map((tab) => (
+            <div
+              className={`EmergencyView_tabs_item${
+                tab.id === selectedTab ? "--active" : ""
```
Comment (dmitrymarchuk) [inline src/pages/EmergencyViewPage/EmergencyViewPage.tsx:240]: use `classNames` lib

File: src/pages/EmergencyViewPage/EmergencyViewPage.tsx [hunk 11]
```diff
@@ -0,0 +1,281 @@
...
+              }`}
+              key={tab.id}
+              onClick={() => updateTab(tab.id)}
```
Comment (dmitrymarchuk) [inline src/pages/EmergencyViewPage/EmergencyViewPage.tsx:243]: `useCallback`. But better extracted into a separate component first not to pollute this one.

File: src/pages/EmergencyViewPage/EmergencyViewPage.tsx [hunk 12]
```diff
@@ -0,0 +1,281 @@
...
+                            isLive={isRoomLive}
+                          />
+                        );
+                      })}
+                    </div>
+                  </>
+                )}
+              </>
+            ))
+          ) : (
```
Comment (dmitrymarchuk) [inline src/pages/EmergencyViewPage/EmergencyViewPage.tsx:274]: can also be extracted

File: src/pages/EmergencyViewPage/EmergencyViewRoom.tsx [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+          src={room.image_url}
+          alt={room.title}
+        />
+        <span>{room.title}</span>
+      </div>
+    </div>
+  );
+};
+
+export default EmergencyViewRoom;
```
Comment (dmitrymarchuk) [inline src/pages/EmergencyViewPage/EmergencyViewRoom.tsx:45]: great small component!

File: src/pages/EmergencyViewPage/EmergencyViewPage.scss [hunk 14]
```diff
@@ -0,0 +1,317 @@
...
+}
+
+.EmergencyView {
+  width: 100%;
+  height: 100%;
+  background-color: opaque-black(0.8);
+  position: absolute;
+  display: flex;
+  flex-direction: column;
+  z-index: 100;
```
Comment (dmitrymarchuk) [inline src/pages/EmergencyViewPage/EmergencyViewPage.scss:214]: I am not sure, but maybe sass function `z` needs to be used here.

File: src/pages/EmergencyViewPage/EmergencyViewPage.scss [hunk 15]
```diff
@@ -0,0 +1,317 @@
+@import "scss/constants.scss";
```
Comment (dmitrymarchuk) [inline src/pages/EmergencyViewPage/EmergencyViewPage.scss:1]: Overall file comments:

- Duplicate styles need to be extracted (together with duplicates in `tsx` file)
- `px`, `%` and other literal values needs to be extracted into constants. If similar alread exist in `constants.scss` those should be picked.
- naming should be changed to conform to BEM convention:
  * `-` character as a separator between words
  * `__` as element indicator
  * `--` as mofificator indicator.

File: src/components/molecules/ScheduleItemNG/ScheduleItemNG.tsx [hunk 16]
```diff
@@ -37,9 +37,17 @@ import "./ScheduleItemNG.scss";
 
 export interface ScheduleItemNGProps {
   event: ScheduledVenueEvent;
+  isShowImage: boolean;
+  isShowBookmark: boolean;
+  isShowDayDate: boolean;
```
Comment (dmitrymarchuk) [inline src/components/molecules/ScheduleItemNG/ScheduleItemNG.tsx:42]: It seems from current usages that these new properties either are all `true` or all `false`. Can be simplified to just one property then.

File: src/components/molecules/ScheduleEventListNG/ScheduleEventListNG.scss [hunk 17]
```diff
@@ -5,6 +5,10 @@
 }
 
 .ScheduleEventSubListNG {
+  &_transparent {
```
Comment (dmitrymarchuk) [inline src/components/molecules/ScheduleEventListNG/ScheduleEventListNG.scss:8]: BEM naming

**Review by dmitrymarchuk — CHANGES_REQUESTED** [review]

**Comment by dmitrymarchuk** [comment]
@alexey-petrov I also checked out a branch but couldn't see the rooms on the Spaces tab on `EmergencyViewPage`. It should be like this but I didn't see any rooms.
image

Can you please provide a reference URL or venue name where the added functionality can be viewed?

**Review by sunny-viktoryia — APPROVED** [review]
going to approve and merge with the thought to finish the review and create a follow-up PR if needed.

**Pull request merged**